### ISSUE TITLE
[Feature Request]: Integrate the R chatgpt package for use in the R-Console or as a extra module

### ISSUE THREAD

**Issue by tomtomme** [jasp-stats/jasp-issues#2010]
### Description

chatgpt support / assistance when coding R in JASP

### Purpose

learn R coding

### Use-case

teaching

### Is your feature request related to a problem?

learning R is hard

### Is your feature request related to a JASP module?

Unrelated

### Describe the solution you would like

Use the existing chatgpt R-Package https://github.com/jcrodriguez1989/chatgpt as a starting point

**Comment by EJWagenmakers** [comment]
Something like this would be cool.

**Comment by loncaricd** [comment]
I know you have many priorities (e.g. I would like to just copy JASP r-code to RStudio and call for something like JASPpackage and reproduce analysis in RStudio or Positron without having to install JASP and ease into reproducibility requirement with such option) but this AI feature could be pure gold for my teaching, my students, and would boost JASP popularity. I know that classical point-and-click statistical software has been developed over the decades with a primary focus on providing a comprehensive and validated set of statistical procedures and their development paradigm has traditionally been different from the agile, API-driven approach of newer (online? cloud?) tools were integrating LLMs is more straightforward. On the other hand, my students are already piping every JASP output into ChatGPT to get their APA section written, so you might as well put a proper API call inside JASP. It could work even for a free, local LLM? Or using packages like gander from @simonpcouch, or ellmer and  chatlas from @tidyverse  and @posit-dev. I know that JASP is not an IDE for coding, have no idea if these packages are any good for interpretation of results (they definitely are for explanation of code so R window would benefit greatly from inclusion of such  modules to explain code. AI in JASP would save my students a copy-pasting step into an outside LLM. Frankly, their prompting force is weak, and integrated access to online or local AI might give them a usable and contextualized baseline that might result in them writing more palatable interpretations of statistical output and some learning alongside. 
In addition to output and R-code window, the same goes for the help files. The current docs are fine if you already know what you're doing, but for a stats n00b like my students, they're cryptic. An AI that can give additional explanations in plain English would be a godsend and "help files" could be even shorter – just ask AI what it means and let it explain to you like to a 5-year old.
Unfortunately, I found just a few models on huggingface that explicitly mention training on R content, like “ajibawa-2023/Code-290k-6.7B-Instruct”, but it could be a starting point, at least for a local inference. There are other great small LLM-s that could do great code continuation and code explanation, although a larger conversational model should be used for help elaboration and results interpretation in APA stile. An API link to some commercial AI provider would be most beneficial for such task because big local models and GPU poor students do not mix well.
This might be a massive win for JASP in everything-AI era. Catering to the copy-paste generation will make JASP the preferred teaching tool everywhere. Students might even accidentally learn something about statisticg (AI in Help), coding (AI in R terminal/window) and proper reporting (AI in Output). Big thumbs-up for this.

**Comment by EJWagenmakers** [comment]
Definitely on our radar!

**Comment by tomtomme** [comment]
@loncaricd 
it needed a year longer, but this is available now
